src/colorizer/Dataset.ts

```diff
@@ -137,27 +139,26 @@ export default class Dataset {
       );
     }
 
-    this.features[name] = {
-      tex: source.getTexture(FeatureDataType.F32),
-      data: source.getBuffer(FeatureDataType.F32),
-      min: source.getMin(),
-      max: source.getMax(),
-      units: metadata?.units || "",
-      type: featureType,
-      categories: featureCategories || null,
-    };
+    return [
+      name,
+      {
+        tex: source.getTexture(FeatureDataType.F32),
+        data: source.getBuffer(FeatureDataType.F32),
+        min: source.getMin(),
+        max: source.getMax(),
+        units: metadata?.units || "",
+        type: featureType,
+        categories: featureCategories || null,
+      },
+    ];
```

Comment:
This was the cause of the feature sorting order bug. It was originally inserting them whenever the async load operation finished, which meant that each completed at different times. Now, each promise returns the resulting array, allowing them to be inserted in the correct order.